projects/packages/videopress/src/class-divi.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Divi integration for Videopress.
+ *
+ * @package automattic/jetpack-videopress
+ **/
+
+namespace Automattic\Jetpack\VideoPress;
+
+/**
+ * Class Divi
+ **/
+class Divi {
+	/**
+	 * The instance.
+	 *
+	 * @var Divi
+	 **/
+	private static $instance = null;
+
+	/**
+	 * Running or not.
+	 *
+	 * @var bool
+	 **/
+	private $running = false;
+
+	/**
+	 * Initializes VideoPress/Divi integration.
+	 *
+	 * Called only once by the Initializer class
+	 *
+	 * @return self
+	 */
+	public static function init() {
+		return self::get_instance()->run();
+	}
+
+	/**
+	 * Get the instance.
+	 *
+	 * @return self
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Run the extension.
+	 *
+	 * @return self
+	 */
+	public function run() {
+		if ( $this->running ) {
+			return $this;
+		}
+
+		$this->running = true;
+		add_action( 'divi_extensions_init', array( $this, 'initialize_extension' ) );
+		return $this;
+	}
+
+	/**
+	 * Creates the extension's main class instance.
+	 */
+	public function initialize_extension() {
+		require_once plugin_dir_path( __FILE__ ) . 'videopress-divi/class-videopress-divi-extension.php';
```

Comment:
Do we need to require the file, given that we're in the package where the classes should be autoloaded?

Reply:
Unfortunately, because it is subclassing a Divi class, we need to ensure Divi is properly loaded, before requiring it:

`add_action( 'divi_extensions_init', array( $this, 'initialize_extension' ) );`